src/herder/test/HerderTests.cpp

```diff
@@ -3801,22 +3752,28 @@ herderExternalizesValuesWithProtocol(uint32_t version)
     REQUIRE(currentALedger() >= currentLedger);
     REQUIRE(currentCLedger() == currentLedger);
 
+    // Arm the upgrade, but don't close the upgrade ledger yet
+    // C won't upgrade until it's on the right LCL
+    upgradeSorobanNetworkConfig(
+        [&](SorobanNetworkConfig& cfg) {
+            cfg.mLedgerMaxTransactionsSizeBytes = 1'000'000;
+            cfg.mTxMaxSizeBytes = 500'000;
+            cfg.mStateArchivalSettings.bucketListWindowSamplePeriod = 1;
+        },
+        simulation, /*applyUpgrade=*/false);
+
     // disconnect C
     simulation->dropConnection(validatorAKey.getPublicKey(),
                                validatorCKey.getPublicKey());
 
+    currentLedger = currentALedger();
+
     // Advance A and B a bit further, and collect externalize messages
     std::map<uint32_t, std::pair<SCPEnvelope, StellarMessage>>
         validatorSCPMessagesA;
     std::map<uint32_t, std::pair<SCPEnvelope, StellarMessage>>
         validatorSCPMessagesB;
 
-    for (auto& node : {A, B, getC()})
-    {
-        // C won't upgrade until it's on the right LCL
-        setupUpgradeAtNextLedger(*node);
```

Comment:
Dima had the same question, so refer back to his last comment where we had a back and forth. TLDR: This test upgraded via direct commit to ltx root, set the upgrade, then disconnected node C. This breaks BucketListDB, so I modified the `upgradeSorobanNetworkConfig` to optionally set the upgrade but not actually apply the upgrade ledger. This should replicate the test correctly, but instead of direct ltx commit, we use loadgen to properly build the upgrade.